### PR TITLE
Add UI elements for plAvLadderMod.

### PR DESCRIPTION
**Pull request by Deledrius**
- Adds dialog for plAvLadderMod.
- Adds icon for plAvLadderMod (and plLadderModifier).
- Adds QVector3, a UI helper for hsVector3.

### CODE REVIEW

File: src/PrpShop/PRP/QVector3.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+#include "QVector3.h"
+
+#include <QLabel>
+#include <QGridLayout>
+
+QVector3::QVector3(QWidget* parent) : QWidget(parent)
+{
+    QGridLayout* layout = new QGridLayout(this);
```
Comment (zrax) [inline src/PrpShop/PRP/QVector3.cpp]: Why not a `QHBoxLayout` since we only use one dimension?

File: src/PrpShop/PRP/Avatar/QAvLadderMod.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+QAvLadderMod::QAvLadderMod(plCreatable* pCre, QWidget* parent)
+    : QCreatable(pCre, kAvLadderMod, parent)
+{
+    plAvLadderMod* mod = plAvLadderMod::Convert(fCreatable);
+
+    fType = new QComboBox(this);
+    fType->addItem("Big", plAvLadderMod::kBig);
+    fType->addItem("FourFeet", plAvLadderMod::kFourFeet);
+    fType->addItem("TwoFeet", plAvLadderMod::kTwoFeet);
```
Comment (zrax) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: Should use `tr()`, and spaces to make it more human readable

File: src/PrpShop/PRP/Avatar/QAvLadderMod.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    layout->addWidget(fType, 1, 1, 1, 2);
+    int typeIdx = fType->findData(mod->getType());
+    if (typeIdx < 0) {
+        fType->addItem(QString("Unknown Value: %1").arg(mod->getType()), mod->getType());
+        fType->setCurrentIndex(fType->count() - 1);
+    }
+    else {
+        fType->setCurrentIndex(typeIdx);
+    }
+    connect(fType, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
```
Comment (zrax) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: Qt 5.7 adds a nicer way to do this...  Can you use something similar to https://github.com/zrax/qfreerdp/blob/master/qfreerdp.h#L25-L34 to make it compatible with Qt 5.7's syntax?

File: src/PrpShop/PRP/Avatar/QAvLadderMod.cpp [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    fEnabled->setChecked(mod->isEnabled());
+
+    layout->addWidget(new QLabel(tr("Type:"), this), 1, 0);
+    layout->addWidget(fType, 1, 1, 1, 2);
+    int typeIdx = fType->findData(mod->getType());
+    if (typeIdx < 0) {
+        fType->addItem(QString("Unknown Value: %1").arg(mod->getType()), mod->getType());
+        fType->setCurrentIndex(fType->count() - 1);
+    }
+    else {
```
Comment (zrax) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: Extra newline

File: src/PrpShop/PRP/QVector3.cpp (hunk 1 above)
Comment (Deledrius) [inline src/PrpShop/PRP/QVector3.cpp]: Short answer: I wasn't aware of it.  Good idea. 👍

File: src/PrpShop/PRP/Avatar/QAvLadderMod.cpp (hunk 3 above)
Comment (Deledrius) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: I'll give it a try.  This was copied from the Qt docs.

File: src/PrpShop/PRP/Avatar/QAvLadderMod.cpp (hunk 4 above)
Comment (Deledrius) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: :trollface: I've now fixed this in Visual Studio's Formatting Options, so hopefully it won't go and ruin these behind my back any more. :wink:

Comment (Hoikas) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: I fail to understand why this is even legal in VS's convoluted brain...

Comment (Deledrius) [inline src/PrpShop/PRP/Avatar/QAvLadderMod.cpp]: The auto-formatting-on-paste option is something I both love and hate.

**Pull request merged**